linera-client/src/chain_listener.rs

```diff
@@ -138,19 +127,20 @@ where
     #[tracing::instrument(level = "trace", skip_all, fields(?chain_id))]
     fn run_with_chain_id<C>(
         chain_id: ChainId,
-        clients: ChainClients<P, S>,
         context: Arc<Mutex<C>>,
-        storage: S,
+        storage: C::Storage,
         config: ChainListenerConfig,
         listening: Arc<Mutex<HashSet<ChainId>>>,
     ) where
-        C: ClientContext<ValidatorNodeProvider = P, Storage = S> + Send + 'static,
+        C: ClientContext + Send + 'static,
+        C::Storage: Clone + Send + Sync + 'static,
+        C::ValidatorNodeProvider: Send + Sync + 'static,
+        <<C::ValidatorNodeProvider as ValidatorNodeProvider>::Node as ValidatorNode>::NotificationStream: Send,
```

Comment:
Aren't these bounds already on `ValidatorNodeProvider`?

Reply:
If not, I think they could/should be (with `cfg(web)`) apropos @ma2bd's recent refactor of the `Local*` traits.